Increased frequency of apy updates

vovunku opened the pull request:
Separated new era handlers on apy updating in newSession and active stackers updating in newEra for relaychain and azero

Checked from 18630602 on kusama to observe two apy updates:

First:
![Screenshot from 2023-07-03 23-24-13](https://github.com/novasamatech/subquery-staking/assets/33938211/59b42b22-e126-4114-b7ea-6a00582d4379)

Second:
![Screenshot from 2023-07-03 23-24-20](https://github.com/novasamatech/subquery-staking/assets/33938211/a4546e7d-88e1-4298-b600-1aea212540c7)


Logs:
[kusama_from_18630602.log](https://github.com/novasamatech/subquery-staking/files/11941217/kusama_from_18630602.log)


Also tested on multichain to avoid errors(and there was a lot! while first era is not started I got different behaviors from different relay chains):

Graphql:
![Screenshot from 2023-07-03 23-21-28](https://github.com/novasamatech/subquery-staking/assets/33938211/1de60753-d493-4bd5-aaae-6ff84553632a)

Logs:
[all_from_1.log](https://github.com/novasamatech/subquery-staking/files/11941229/all_from_1.log)

ERussel reviewed (changes requested):
As we discussed:

1) Make sure we only recalculate APY each session and cache's validator on StakersElected
2) Make sure that we also recalculate APY when new stakers elected

vovunku commented:
1. fixed with separating by two handlers
2. fixed by updating apy on new era also

Should bee good now.

> As we discussed:
> 
> 1. Make sure we only recalculate APY each session and cache's validator on StakersElected
> 2. Make sure that we also recalculate APY when new stakers elected

ERussel commented on src/mappings/kusama.ts:
```diff
@@ -20,6 +20,17 @@ export async function handleKusamaNewEra(_: SubstrateEvent): Promise<void> {
 }
 
+export async function handleKusamaNewSession(_: SubstrateEvent): Promise<void> {
+    let validatorEraInfoDataSource = new ValidatorEraInfoDataSource();
```
As we are recreating datasource on each session that means also we are fetching validators each session. This is time-consuming operation. Better fetch validators once in era and then cache.

vovunku commented:
Similar testes:

From zero:

![Screenshot from 2023-07-07 17-46-46](https://github.com/novasamatech/subquery-staking/assets/33938211/72bef405-ec66-4770-a32d-5055e6824a01)

Logs:
[all_from_1.log](https://github.com/novasamatech/subquery-staking/files/11983872/all_from_1.log)

Note: as we made broken logic for relaychains on first blocks **I saw 1500. APY on polkadot relaychain**(but forgot made screenshot)

From 18685151 kusama:

Before:
![Screenshot from 2023-07-07 17-54-45](https://github.com/novasamatech/subquery-staking/assets/33938211/d296544d-cbd2-402c-a20d-dbcc3947d568)
After:
![Screenshot from 2023-07-07 17-55-32](https://github.com/novasamatech/subquery-staking/assets/33938211/83cf5135-eb46-4bf6-8894-4bd3b4556be5)

Logs:

[kusama_from_18685151.log](https://github.com/novasamatech/subquery-staking/files/11983894/kusama_from_18685151.log)

vovunku commented:
![Screenshot from 2023-07-11 09-49-19](https://github.com/novasamatech/subquery-staking/assets/33938211/85ba0a27-ad16-43d8-8f52-31a34eacde2e)

[tmp.log](https://github.com/novasamatech/subquery-staking/files/12012108/tmp.log)


I can make more tests, if it is necessary!

vovunku commented:
Similar test for kusama session update
First session:
![Screenshot from 2023-07-11 09-59-21](https://github.com/novasamatech/subquery-staking/assets/33938211/9849f17c-98bd-497f-a194-507e87a53b77)


Second session:
![Screenshot from 2023-07-11 10-00-14](https://github.com/novasamatech/subquery-staking/assets/33938211/b648a335-6c9c-4f05-8ddc-66f4205d344c)

Log:
[tmp.log](https://github.com/novasamatech/subquery-staking/files/12012204/tmp.log)
